crates/solidity/outputs/typescript/lib/src/generated/cst.rs

```diff
@@ -23,10 +23,17 @@ impl Node {
     pub fn rule(kind: RuleKind, children: Vec<Option<Rc<Self>>>) -> Option<Rc<Self>> {
         let children: Vec<_> = children.into_iter().filter_map(|e| e).collect();
         if children.is_empty() {
-            None
-        } else {
-            Some(Rc::new(Self::Rule { kind, children }))
+            return None;
+        }
+        if children.len() == 1 {
```

Comment:
Nit: I recently learned that descructuring ranges with a specific length is supported:

```rust
if let [Self::Group { children }] = children[..] {
  return Some(Rc::new(Self::Rule {
    kind,
    children,
  }));
}
```

I think it would also allow using `children` directly without cloning. Thoughts?